src/components/BShuttle/BShuttle.vue

```diff
@@ -0,0 +1,486 @@
+      selectedOptions: {'available': [], 'chosen': []},
+      chosenOptions: (this.value && this.options) ? this.setOptions(this.options, this.value, 'chosen') : [],
+      lastClick: 0,
+      id: null,
+      allToChosen: require('../../../static/doubleArrow.svg'),
+      allToAvailable: require('../../../static/doubleArrow-left.svg')
+    }
+  },
+
+  created() {
+    this.$validator.extend('notEmpty', {
+      getMessage: field => `The ${field} input is required.`,
+      validate: value => value.length > 0
+    });
+    this.$validator.attach({
+      name: this.validationId,
+      rules: 'notEmpty',
+      alias: this.chosenLabel
+    });
+
+  },
+
+  mounted() {
+    this.id = this._uid
+  },
+
+  methods: {
+
+    /**
+     * @param {Array<Object>} options
+     * @param {Array<Number>} selected
+     * @param {String} type
+     * this method sets up the initial state of the options
+     */
+    setOptions: function (options, selected, type) {
+      const availableOptions = []
+      const chosenOptions = []
+      options.forEach(function (option) {
+        if ((selected === null) || (selected.indexOf(option.id) === -1)) {
+          availableOptions.push(option)
+        } else {
+          chosenOptions.push(option)
+        }
+      })
+      if (type === 'available') {
+        return this.sortById(availableOptions)
+      } else if (type === 'chosen') {
+        return this.sortById(chosenOptions)   
+      }
+    },
+
+    /**
+     * @param {Object} option
+     * @param {String} optionType
+     * this determines if an option has been highlighted and
+     * sets the correct class.
+     */
+    optionsClass: function (option, optionType) {
+      return {
+        'b-shuttle-options': true,
+        'selected': this.selectedOptions[optionType].indexOf(option.id) > -1
+      }
+    },
+
+    /**
+     * @param {Object} option
+     * @param {String} optionType
+     * @param {String} otherOptionType
+     * @param {Object} event
+     * this is fired when a line is clicked and updates
+     * selectedOptions accordingly.
+     */
+    onOptionClick (option, optionType, otherOptionType, event) {
+      if (this.selectedOptions[otherOptionType].length > 0) { this.selectedOptions[otherOptionType] = [] }
+      if (event.shiftKey && (this.lastClick > 0)) {
+        this.multipleSelect(option, optionType)
+      } else if (event.altKey) {
+        if (optionType === 'available') {
+          const index = this.availableOptions.findIndex(item => item.id === option.id)
+          const optionToMove = this.availableOptions[index]
+          this.availableOptions.splice(index, 1)
+          this.moveItems(this.availableOptions, [optionToMove], optionType)
+        } else {
+          const index = this.chosenOptions.findIndex(item => item.id === option.id)
+          const optionToMove = this.chosenOptions[index]
+          this.chosenOptions.splice(index, 1)
```

Comment:
This could be a nice as callback for `moveItems`

Reply:
what benefit would we gain from that?